pkg/machines/components/networks/networkList.jsx

```diff
@@ -45,8 +56,7 @@ export class NetworkList extends React.Component {
     render() {
         const { dispatch, networks, resourceHasError, onAddErrorNotification, vms, nodeDevices, interfaces } = this.props;
```

Comment:
nodeDevices && interfaces  are not used any more, in the way you implemented it. The getNetworkDevices function that you keept (from the two we had) is asynchronous, the one your dropped was not asynchronous. My concern with the asynchronous version used here, is that you might open the dialog before the state is set, resulting in the devices being reset in the dialog through the props while that is open.
For that reason, I would rather see the other version of the function being kept, unless there is a reason behind this decision.

Reply:
Looking at it, I noticed neither is correct, as neither gives a full list of devices we could use.
The synchronous function uses combination of 2 apis (virsh nodedev-list and virsh iface-list). I prefer to stop using that anywhere we can, because it's not ideal. It won't list all network devices (for example it won't list a list of interface created by $brctl addbr). Also a lot of libvirt network related APIs (liek virsh iface-bridge) are just broken. So that's why I'm pushing for using alternatives if possible.
The asynchronous call there (using $ls /sys/class/net) is also not ideal, as it doesn't give any closer information about the device/bridge.
So I'm thinking about perhaps using a NetworkManager there. We also use it in machines already (to figure out if to provide a link to network page for a network device). We could use NetworkManager to get a list of all network devices and bridges. I already talked a bit with @mvollmer about possilibity to create a shared JS library d-bus networkamanger calls. 
It would however either require to bring networkmanager as a dependency or we use networkmanager defaultly, and if it's not present on a system, use the old libvirt calls as a fallback. WDYT?

Reply:
networkmanager as recommended dependency and disabling opening the dialogs if the package is missing seems ok to me, (same as we do for virt-install)
but I would like that we discuss it shorlty with pitti before you move forward.